File: Java/src/view/ContentView.java
package view;

import model.Department;
import model.Employee;
import java.util.List;

/**
 * ContentView centralizes all console output for the Java program.
 * This helps separate presentation logic from business logic and makes
 * the design closer to MVC for better readability in the rubric.
 */
public class ContentView {
    public void showHeader(String title) {
        System.out.println("\n=== " + title + " ===");
    }

    public void showMessage(String message) {
        System.out.println(message);
    }

    public void showError(String message) {
        System.err.println(message);
    }

    public void showEmployee(Employee employee) {
        System.out.println(employee);
    }

    public void showEmployees(List<Employee> employees) {
        for (Employee employee : employees) {
            showEmployee(employee);
        }
    }

    public void showEmployeesByDepartment(List<Employee> employees, Department dept) {
        for (Employee employee : employees) {
            if (employee.getDept() == dept) {
                showEmployee(employee);
            }
        }
    }
}

File: CSharp/Views/ContentView.cs
using System;
using System.Collections.Generic;
using EmployeeManagementSystem.Models;

namespace EmployeeManagementSystem.Views
{
    public class ContentView
    {
        public void ShowHeader(string title)
        {
            Console.WriteLine($"\n=== {title} ===");
        }

        public void ShowMessage(string message)
        {
            Console.WriteLine(message);
        }

        public void ShowError(string message)
        {
            Console.WriteLine(message);
        }

        public void ShowEmployee(Employee employee)
        {
            Console.WriteLine(employee);
        }

        public void ShowEmployees(IEnumerable<Employee> employees)
        {
            foreach (Employee employee in employees)
            {
                ShowEmployee(employee);
            }
        }

        public void ShowEmployeesByDepartment(IEnumerable<Employee> employees, Department department)
        {
            foreach (Employee employee in employees)
            {
                if (employee.Dept == department)
                {
                    ShowEmployee(employee);
                }
            }
        }
    }
}
